### ISSUE TITLE
$dummyValue placeholder conflict with stylelint rules

### ISSUE THREAD

**Issue by sboudrias-aa** [styled-components/stylelint-processor-styled-components#81]
When writing styled-components looking like this:

```js
const Row = styled.div`
    background: #ccc;
`;

const Cell = styled.div`
    ${Row}:hover {
        color: red;
    }
`;
```

Stylelint will report two issues:

> 34:5  ✖  Unexpected unknown type selector "$dummyValue"   selector-type-no-unknown
> 34:5  ✖  Expected "$dummyValue" to be "$dummyvalue"       selector-type-case

The `selector-type-no-unknown` error is easily fixable by fixing our config:

```js
{
    'selector-type-no-unknown': [true, { ignoreTypes: ['$dummyValue'] }]
}
```

But the `selector-type-case` doesn't have "easy" workaround: https://stylelint.io/user-guide/rules/selector-type-case/ (apart from turning the rule off).

I'm wondering is maybe we could replace `$dummyValue` with `$dummyvalue`?

**Comment by emilgoldsmith** [comment]
Hi @sboudrias-aa, I would deem this as a duplicate of #34, which we are on our way to making an API change that would allow you to handle these cases through #68 and #70, So for now I don't think we'll change the `$dummyValue` to all lowercase simply because we'll have a complete fix soon and also it wouldn't fix the first of the two lint errors anyway.

I'll close this issue now because it's a duplicate, and hopefully we'll soon push the code to fix this, though it just has to be written first. Thanks for your patience!

**Comment by ViggoV** [comment]
I have this issue here a year and a half later, also specifically the `$dummyValue` placeholder issue though only the first of the two mentioned above. Interestingly it only occurs in one instance, whereas the pattern that seems to trigger it occurs throughout the project. It triggers on `props.active`

```
${props => props.active && css`
  background-color: ${props.theme.biqBlue};  
  color: white;
`}
```

stylelint-processor-styled-components@1.5.1

**Comment by chinesedfan** [comment]
@ViggoV The processor can not always parse interpolations with right things. But you can use [interpolation-tagging](https://www.styled-components.com/docs/tooling#interpolation-tagging) to help it. If you have ideas to make it more intelligent, feel free to send a PR or share your solution by an new issue.

```
const Cell = styled.div`
    ${/* sc-selector */ Row}:hover {
        color: red;
    }
`;
```

**Comment by ViggoV** [comment]
Interesting, thanks for the link. However my use case doesn't seem to be covered in the examples since it is a conditional rather than an interpolation of an existing css concept. `/* sc-declaration */` seems to do the trick though. :)

**Comment by emilgoldsmith** [comment]
The tags are also mostly meant as an "override". So if `sc-declaration` works for you that's perfect. Otherwise there's always `sc-custom` for anything the normal ones don't cover. Tags are basically for anything we can't evaluate statically
